### PR TITLE
allow review endpoints on missing namespaces

### PR DESCRIPTION
**Pull request by deads2k**
fixes https://github.com/openshift/origin/issues/11300

### CODE REVIEW

File: vendor/k8s.io/kubernetes/plugin/pkg/admission/namespace/lifecycle/admission.go [hunk 1]
```diff
@@ -205,6 +205,7 @@ var accessReviewResources = map[unversioned.GroupResource]bool{
 	unversioned.GroupResource{Group: "", Resource: "resourceaccessreviews"}:      true,
 	unversioned.GroupResource{Group: "", Resource: "localresourceaccessreviews"}: true,
 	unversioned.GroupResource{Group: "", Resource: "selfsubjectrulesreviews"}:    true,
+	unversioned.GroupResource{Group: "", Resource: "subjectrulesreviews"}:        true,
```
Comment (deads2k) [inline vendor/k8s.io/kubernetes/plugin/pkg/admission/namespace/lifecycle/admission.go]: @mfojtik we need to fix the need to add to two lists.

**Comment by deads2k** [comment]
[test]

File: test/cmd/projects.sh [hunk 2]
```diff
@@ -3,6 +3,21 @@ source "$(dirname "${BASH_SOURCE}")/../../hack/lib/init.sh"
 trap os::test::junit::reconcile_output EXIT
 
 os::test::junit::declare_suite_start "cmd/projects"
+
+os::test::junit::declare_suite_start "cmd/projects/lifecycle"
+# resourceaccessreview
+os::cmd::expect_success 'oc policy who-can get pods -n missing-ns'
```
Comment (stevekuznetsov) [inline test/cmd/projects.sh]: This is a weak test

File: test/cmd/projects.sh [hunk 3]
```diff
@@ -3,6 +3,21 @@ source "$(dirname "${BASH_SOURCE}")/../../hack/lib/init.sh"
...
+# selfsubjectaccessreview
+os::cmd::expect_success 'oc policy can-i get pods -n missing-ns'
+# selfsubjectrulesreivew
+os::cmd::expect_success 'oc policy can-i --list -n missing-ns'
```
Comment (stevekuznetsov) [inline test/cmd/projects.sh]: This is a weak test

**Comment by deads2k** [comment]
> Not a fan of naked os::cmd::expect_success.

Non-zero return checks exactly what need. Did this fail?  No?  Ok, success.  The value returned doesn't matter.

**Comment by stevekuznetsov** [comment]
Why not check that the rules review correctly finds nothing for the missing namespace? The CLI could exit with success and do the wrong thing ...

**Comment by deads2k** [comment]
> Why not check that the rules review correctly finds nothing for the missing namespace? The CLI could exit with success and do the wrong thing

Because it in most cases, the answer isn't nothing.

**Comment by deads2k** [comment]
yum re[test]

**Comment by deads2k** [comment]
https://ci.openshift.redhat.com/jenkins/job/test_pr_origin/9920/ flaked on https://github.com/openshift/origin/issues/11240 re[test]

File: pkg/project/admission/lifecycle/admission.go [hunk 4]
```diff
@@ -33,10 +33,17 @@ type lifecycle struct {
...
 }
 
-var recommendedCreatableResources = sets.NewString("resourceaccessreviews", "localresourceaccessreviews")
+var recommendedCreatableResources = map[unversioned.GroupResource]bool{
+	authorizationapi.Resource("resourceaccessreviews"):      true,
+	authorizationapi.Resource("localresourceaccessreviews"): true,
+	authorizationapi.Resource("subjectaccessreviews"):       true,
+	authorizationapi.Resource("localsubjectaccessreviews"):  true,
+	authorizationapi.Resource("selfsubjectrulesreviews"):    true,
+	authorizationapi.Resource("subjectrulesreviews"):        true,
```
Comment (liggitt) [inline pkg/project/admission/lifecycle/admission.go]: `subjectrulesreviews` isn't in the upstream set of `accessReviewResources`, need a carry commit to add it there

**Comment by deads2k** [comment]
@mfojtik look good to you otherwise?  if so, I'll update and merge.

**Comment by mfojtik** [comment]
@deads2k looks good

**Comment by deads2k** [comment]
[merge]

**Comment by deads2k** [comment]
yum re[test] re[merge]

**Comment by deads2k** [comment]
re[test]

**Pull request merged**